polaris.shopify.com/src/components/IconGrid/IconGrid.tsx

```diff
@@ -1,76 +1,57 @@
-  true,
-  /\.svg$/
-);
 import styles from "./IconGrid.module.scss";
-import { Icon, SearchResultItem } from "../../types";
+import iconMetadata from "@shopify/polaris-icons/metadata";
+import Link from "next/link";
 
-interface IconGridProps {
-  children: React.ReactNode;
+interface Props {
+  title?: string;
+  icons: typeof iconMetadata;
+  activeIcon?: string;
+  query?: string;
 }
 
-function IconGrid({ children }: IconGridProps) {
-  return <ul className={styles.IconGrid}>{children}</ul>;
-}
-
-interface IconGridItemProps extends SearchResultItem {
-  icon: Icon;
-  onClick: (iconName: string) => void;
-  isSelected?: boolean;
-}
-
-function IconGridItem({
-  icon,
-  onClick,
-  isSelected,
-  searchResultData,
-}: IconGridItemProps) {
+function IconGrid({ title, icons, activeIcon, query = "" }: Props) {
   return (
-    <li
-      key={`${icon.name}+${icon.set}`}
-      className={className(
-        styles.Icon,
-        searchResultData?.isHighlighted && styles.isHighlighted,
-        isSelected && styles.isSelected
-      )}
-      {...searchResultData?.itemAttributes}
-      id={icon.fileName}
-    >
-      <Tooltip
-        ariaLabel={icon.description}
-        placement="top"
-        renderContent={() => (
-          <div>
-            <p>
-              {icon.description === "N/A" ? "No description" : icon.description}
-            </p>
-          </div>
-        )}
-      >
-        <button
-          onClick={() => onClick(icon.name)}
-          tabIndex={searchResultData?.tabIndex}
-        >
-          <div className={styles.SVGWrapper}>
-            <Image
-              src={importedSvgs(`./${icon.fileName}.svg`)}
-              alt={icon.description}
-              width={20}
-              height={20}
-              fadeIn={false}
-            />
-          </div>
-          <span style={{ fontSize: 12, color: "#aaa" }}>{icon.name}</span>
-        </button>
-      </Tooltip>
-    </li>
+    <>
+      {title ? <h2 className={styles.SectionHeading}>{title}</h2> : null}
+      <ul className={styles.IconGrid}>
+        {Object.keys(icons).map((iconFileName) => (
+          <li key={iconFileName}>
+            <Link
+              href={{
+                pathname: "/icons",
+                query: {
+                  icon: iconFileName,
+                  ...(query === "" ? {} : { q: query }),
+                },
+              }}
+              scroll={false}
+            >
+              <a
+                className={className(
+                  styles.Icon,
+                  activeIcon === iconFileName && styles.IconSelected
+                )}
+              >
+                <div className={styles.ImageWrapper}>
+                  <Image
+                    src={`/icons/${iconFileName}.svg`}
+                    alt={icons[iconFileName].description}
+                    width={20}
+                    height={20}
+                    fadeIn={false}
+                    title={icons[iconFileName].name}
```

Comment:
This `title` doesn't add much value IMO. Its value is the same as the text next to the icon.

Worth discussing, but I kinda miss the tooltips when hovering. They made it easy to get the description of an icon without clicking into it and having to move my eyes to the sidebar.